### PR TITLE
Implement availablilty items/holdings. BBS-28 BBS-29

### PR DESCRIPTION
**Pull request by tripox**

### CODE REVIEW

File: modules/aleph/includes/aleph.availability.inc [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  }
+
+  return $result;
+}
+
+/**
+ * Implements provider availability, holdings.
+ *
+ * @param array $provider_ids
+ *   Array of ting object ids (faust).
```
Comment (danquah) [inline modules/aleph/includes/aleph.availability.inc]: is it called faust in Aleph?

File: modules/aleph/includes/aleph.availability.inc [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    $material->setId($provider_id);
+    $items = $material_handler->getHoldings($material);
+
+    $reserved_count = 0;
+    $available_count = 0;
+
+    foreach ($items as $item) {
+      // Count the total amount on sub-libraries and see if it's available or
+      // reservable.
+      if (array_key_exists($item->getSubLibraryCode(), $sub_libraries)) {
```
Comment (danquah) [inline modules/aleph/includes/aleph.availability.inc]: No chance of getSubLibraryCode() returning null?

File: modules/aleph/src/Aleph/AlephMaterial.php [hunk 3]
```diff
@@ -114,6 +121,145 @@ public function isRenewable() {
...
+   *    The type of material (z30-material).
+   */
+  public function setType($type) {
+    $this->type = $type;
+  }
+
+  /**
+   * Get the type of material.
+   *
+   * @return string
```
Comment (danquah) [inline modules/aleph/src/Aleph/AlephMaterial.php:156]: Consider adding examples from real data in the description of all the non-trivial getters. It is a bit of work, but very much worth it for the next person who looks at using the getter.

File: modules/aleph/includes/aleph.availability.inc [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        $sub_libraries[$item->getSubLibraryCode()]['total_count']++;
+        $sub_libraries[$item->getSubLibraryCode()]['available'] = $item->isAvailable();
+        $sub_libraries[$item->getSubLibraryCode()]['reservable'] = $item->isReservable();
+      }
+      else {
+        $sub_libraries[$item->getSubLibraryCode()] = array(
+          'placement' => array($item->getSubLibrary(), $item->getCollection()),
+          // Reference count not available in Aleph.
```
Comment (danquah) [inline modules/aleph/includes/aleph.availability.inc:82]: What is a reference-count?

File: modules/aleph/src/Aleph/AlephMaterial.php [hunk 5]
```diff
@@ -127,13 +273,38 @@ public static function loansFromBorInfo(\SimpleXMLElement $xml) {
...
   }
 
+  /**
+   * Create material from item.
+   *
+   * @param $item
+   *
+   * @return \Drupal\aleph\Aleph\AlephMaterial
+   */
+  public static function materialFromItem(\SimpleXMLElement $item) {
```
Comment (danquah) [inline modules/aleph/src/Aleph/AlephMaterial.php:294]: Should we add some example xml-documents to the repo? Or maybe just include examples in the source-code? It is very hard to form an opinion on whether $item->xpath('z30/z30-sub-library')[0]); is the best way to query for the data or not ...

**Review by danquah — CHANGES_REQUESTED** [review]

File: modules/aleph/includes/aleph.availability.inc (hunk 4 above)
Comment (tripox) [reply modules/aleph/includes/aleph.availability.inc:82]: Not sure, but there's nothing like it available in Aleph and they didn't seem to know what it was and not all providers implements it.

**Review by danquah — APPROVED** [review]
No idea why those tests are suddenly failing - the commits leading up to them failing does not seem to be related, so approving

**Pull request merged**